src/app/modules/table2/table-header/table-header.component.html

```diff
@@ -1,42 +1,43 @@
 <ng-container [ngSwitch]="tableHeaderModel.type">
-    <ng-container *ngSwitchCase="TableHeaderTypeEnum.SORT">
-        <app-button
-            [title]="tableHeaderModel.label"
-            [ariaLabel]="'Activate to sort'"
-            [buttonID]="'header_' + tableHeaderModel.columnKey"
-            (callBack)="headerClickEvent($event)"
-            buttonType="ds-u-padding--0 ds-u-border--0 ds-u-color--base"
-        >
-            {{ tableHeaderModel.label }}
-            <fa-icon 
-                *ngIf="tableHeaderModel.sort !== TableHeaderSortEnum.NONE" 
-                class="ds-u-display--inline-block"
-                [icon]="tableHeaderModel.sort === TableHeaderSortEnum.ASC ? faCaretUp : faCaretDown"
-            ></fa-icon>
-        </app-button>
-    </ng-container>
+  <ng-container *ngSwitchCase="TableHeaderTypeEnum.SORT">
+    <app-button
+      [title]="tableHeaderModel.label"
+      [ariaLabel]="'Activate to sort'"
+      [buttonID]="'header_' + tableHeaderModel.columnKey"
+      (callBack)="headerClickEvent($event)"
+      buttonType="ds-u-padding--0 ds-u-border--0 ds-u-color--base"
+    >
+      {{ tableHeaderModel.label }}
+      <fa-icon
+        *ngIf="tableHeaderModel.sort !== TableHeaderSortEnum.NONE"
+        class="ds-u-display--inline-block"
+        [icon]="tableHeaderModel.sort === TableHeaderSortEnum.ASC ? faCaretUp : faCaretDown"
+      ></fa-icon>
+    </app-button>
+  </ng-container>
 
-    <ng-container *ngSwitchCase="TableHeaderTypeEnum.CHECKBOX">
-        <app-choice
-            [label]="tableHeaderModel.label"
-            [value]="tableHeaderModel.columnKey"
-            [isChecked]="tableHeaderModel.isChecked"
-            [showTitle]="false"
-            size="small"
-            (choiceChange)="headerClickEvent($event)"
-        >
-        </app-choice>
-    </ng-container>
+  <ng-container *ngSwitchCase="TableHeaderTypeEnum.CHECKBOX">
+    <app-choice
+      [label]="tableHeaderModel.label"
+      [value]="tableHeaderModel.columnKey"
+      [isChecked]="tableHeaderModel.isChecked"
+      [showTitle]="true"
+      size="small"
+      (choiceChange)="headerClickEvent($event)"
+    >
+    </app-choice>
+    <div [class]="'sr-only'" [title]="tableHeaderModel.label" tabindex="0">
```

Comment:
i think for this to be compliant, the label has to be in a <label> element with a for attribute equal to the id of the checkbox.  I think you were onto something before with the labelClass="sr-only", did that cause an issue that couldn't be fixed?

Reply:
I removed it from the choice component because it was causing the checkbox to be cut off, however, removing it didn't cause the label to actually appear, so I think we are good there. I think here, the header isn't showing at all, so I was adding this element in the container to show in the header as opposed to the choice. Does that make sense? In that case will this work? @joshglazer

Reply:
not show in the header, but be present for the reader